robotoff/workers/tasks/update_recycle.py

```diff
@@ -0,0 +1,206 @@
+
+logger = get_logger(__name__)
+
+
+def update_recycling(username, password):
+    """
+    Function to update "Recycle" image for the product
+    based on triggers
+
+    :param username:
+    :param password:
+    :return:
+    """
+
+    recycling_triggers = {
+        "en": ["throw away", "recycle"],
+        "fr": ["consignesdetri.fr", "recycler", "jeter", "bouteille"],
+    }
+    # get products dataset
+    dataset = ProductDataset.load()
+
+    # iterate products
+    for product in dataset.stream().filter_nonempty_text_field("code"):
+
+        if "packaging-photo-to-be-selected" in product.get("states", ""):
+            images = get_images(product.get("code"))
+            if images:
+                images_ids = [
+                    i
+                    for i, j in images.get("product", {}).get("images", {}).items()
+                    if not j.get("imgid")
+                ]
+                pack_images = {
+                    i: j
+                    for i, j in images.get("product", {}).get("images", {}).items()
+                    if "packaging" in i
+                }
+                images_ids = list(set(images_ids))
+
+                for i in images_ids:
+                    # imageid - i, product
+                    for lang in recycling_triggers.keys():
+
+                        if check_image_in_pack(i, lang, pack_images):
+                            continue
+
+                        select_image(
+                            product.get("code"),
+                            i,
+                            product,
+                            lang,
+                            recycling_triggers[lang],
+                            pack_images,
+                            username,
+                            password,
+                        )
+
+
+def get_images(ean):
+    """
+    Get images for the product
+
+    :param ean:
+    :return:
+    """
+    url = (
+        "https://world.openfoodfacts.org/api/v0/product/" + ean + ".json?fields=images"
+    )
+    try:
+        r = requests.get(url)
+        if r.ok:
+            return r.json()
+    except Exception:
+        pass
+    return {}
+
+
+def select_image(
+    ean, image, product, lang, recycling_triggers, pack_images, username, password
+):
+    """
+    Find "Recycle" image and select for the product
+
+    :param ean:
+    :param image: imgid
+    :param product: object from dataset
+    :param lang: en/fr/...
+    :param recycling_triggers: ["throw away", "recycle"]
+    :param pack_images_keys: dict_keys to check if unselect is needed
+    :param username:
+    :param password:
+    :return:
+    """
+    field = "packaging_{}".format(lang)
+
+    data = get_json_for_image(product.get("code"), image)
+    image_text = data.get("responses", [])
+    if image_text:
+        image_text = image_text[0].get("fullTextAnnotation", {}).get("text", "")
+
+        if any([1 if i in image_text.lower() else 0 for i in recycling_triggers]):
```

Comment:
This could be better written as:

```suggestion
        if any(trigger in image_text.lower() for trigger in recycling_triggers):
```

Reply:
Ok